pandas/core/strings/object_array.py

```diff
@@ -339,14 +339,8 @@ def _str_split(
             new_pat: str | re.Pattern
             if regex is True or isinstance(pat, re.Pattern):
                 new_pat = re.compile(pat)
-            elif regex is False:
-                new_pat = pat
-            # regex is None so link to old behavior #43563
             else:
-                if len(pat) == 1:
-                    new_pat = pat
-                else:
-                    new_pat = re.compile(pat)
+                new_pat = pat
```

Comment:
I think we want to keep this, otherwise it would not be a bugfix for `pd.ArrowDtype(pa.string())` but changing behaviour for all other string dtypes?

Reply:
As discussed above https://github.com/pandas-dev/pandas/pull/58418#discussion_r1582205493, the bug is caused by that this logic is not implemented in the `_str_split` method of `pd.ArrowDtype(pa.string())`. To fix it, and to make two `_str_split` implementations look more alike, I moved this logic outside before calling `_str_split`. So I think the behaviors for other dtypes have not been changed.

Moreover I think the existing tests have covered all combinations of parameters, and as long as they all pass, the behaviors should still be the same.